Added support for short, short[], byte & byte[]

skigrinder opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
Added support for short, short[], byte & byte[].
Added testing for these types. 

## Motivation and Context
- Improves functionality.
- Closes nanoframework/Home#622.

## How Has This Been Tested?<!-- (if applicable) -->
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots<!-- (if appropriate): -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

skigrinder commented:
Added support for short, short[], byte & byte[].
Added testing for these types

skigrinder commented:
Thanks Jose!

On 2020-07-04 11:34 AM, José Simões wrote:
>
> Reopened #23 
> <https://github.com/nanoframework/lib-nanoFramework.Json/pull/23>.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nanoframework/lib-nanoFramework.Json/pull/23#event-3512723398>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AO2PVOOGDZR4ZPFGNCIBIB3RZ5R37ANCNFSM4OQPP2SQ>.
>

josesimoes commented:
@skigrinder seems that your working branch is behind develop. Care to rebase please?

skigrinder commented:
Hi Jose,
I'm reading up on the rebase procedure and don't want to screw things up.
Is it correct to rebase from my current branch (type-additions) onto the 
origin/develop branch?
 From VS, is this what needs to be done or have I misunderstood the 
rebase process?



On 2020-07-04 3:52 PM, José Simões wrote:
>
> @skigrinder <https://github.com/skigrinder> seems that your working 
> branch is behind develop. Care to rebase please?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nanoframework/lib-nanoFramework.Json/pull/23#issuecomment-653816517>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AO2PVOMEDQBBFUSCMPXS7KTRZ6QC3ANCNFSM4OQPP2SQ>.
>

josesimoes commented:
Not sure what's your "upstream" repo. It can be "origin" as you have it... The branch it's OK: develop.
Basically you want to pull all the new stuff and commit your changes on top of it.

For the next iteration, make sure you have your develop branch in sync with upstream. Before you branch, that is. 😉

skigrinder commented:
Hi Jose,
Did the rebase and synced everything into the existing PR.
Sorry for the confusion.  I created this fork shortly before you did a 
bunch of new updates & NuGet releases.
In retrospect, I should have clobbered the fork and created a new one 
but it was good to go through the rebase process.
Hoping I did this correctly.
Can you please take a look and let me know if I need to do anything more?
Thanks,
    Jim

On 2020-07-05 12:29 PM, José Simões wrote:
>
> Not sure what's your "upstream" repo. It can be "origin" as you have 
> it... The branch it's OK: develop.
> Basically you want to pull all the new stuff and commit your changes 
> on top of it.
>
> For the next iteration, make sure you have your develop branch in sync 
> with upstream. Before you branch, that is. 😉
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nanoframework/lib-nanoFramework.Json/pull/23#issuecomment-653922660>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AO2PVOKGBNOEYHZFJSFPTT3R2DBCFANCNFSM4OQPP2SQ>.
>

josesimoes commented:
Yes looks goo now. Thanks!

Now there is a bug reported by sonarcloud. Care to take a look and fix?

skigrinder commented:
Hi Jose',
Found and fixed the issue identified by sonarcloud.
Thanks,
    Jim


On 2020-07-05 2:29 PM, José Simões wrote:
>
> Yes looks goo now. Thanks!
>
> Now there is a bug reported by sonarcloud. Care to take a look and fix?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nanoframework/lib-nanoFramework.Json/pull/23#issuecomment-653935329>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AO2PVOK3S7ARZZIWNTKJMNLR2DPDDANCNFSM4OQPP2SQ>.
>

networkfusion commented on source/nanoFramework.Json.sln:
```diff
@@ -1,42 +1,35 @@
-		NuGet.Config = NuGet.Config
-		version.json = version.json
-	EndProjectSection
+Project("{11A8DD76-328B-46DF-9F39-F559912D0360}") = "nanoFramework.Json", "nanoFramework.Json\nanoFramework.Json.nfproj", "{181141FB-0D05-41C8-8692-81A2989E6B4B}"
```
Nuget Spec is required, not sure why it has been deleted...

networkfusion commented on source/nanoFramework.Json.Test/app.config:
```diff
@@ -0,0 +1,11 @@
+    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+      <dependentAssembly>
+        <assemblyIdentity name="mscorlib" publicKeyToken="c07d481e9758c731" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-1.7.3.5" newVersion="1.7.3.5" />
```
app.config files are bad, it shows that there is an issue with the nugets installed. Try removing the file and ensure it still builds after updating the nugets.

networkfusion commented on source/nanoFramework.Json.Test/nanoFramework.Json.Test.nfproj:
```diff
@@ -28,17 +28,21 @@
     </Reference>
   </ItemGroup>
   <ItemGroup>
+    <None Include="app.config" />
```
app.config files are bad, it shows that there is an issue with the nugets installed. Try removing the file and ensure it still builds after updating the nugets.

networkfusion commented on source/nanoFramework.Json/app.config:
```diff
@@ -0,0 +1,11 @@
+    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+      <dependentAssembly>
+        <assemblyIdentity name="mscorlib" publicKeyToken="c07d481e9758c731" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-1.7.3.5" newVersion="1.7.3.5" />
```
app.config files are bad, it shows that there is an issue with the nugets installed. Try removing the file and ensure it still builds after updating the nugets.

networkfusion commented on source/nanoFramework.Json/nanoFramework.Json.nfproj:
```diff
@@ -43,21 +43,26 @@
     </Reference>
   </ItemGroup>
   <ItemGroup>
+    <None Include="app.config" />
```
app.config files are bad, it shows that there is an issue with the nugets installed. Try removing the file and ensure it still builds after updating the nugets.

skigrinder commented:
Hi Robin,
Thanks for the help.
I removed the app.config files, cleaned up the extraneous whitespace and 
rebuilt everything.
Please let me know if you see any other issues.
Thanks,
    Jim

On 2020-07-06 3:51 AM, Robin Jones wrote:
>
> *@networkfusion* commented on this pull request.
>
> White space changes make it difficult to review the changes (these 
> should be done in a seperate PR...) so I think I have caught all of 
> the problems. app.config files are bad and nF doesnt support them, so 
> they need to be removed and resolved before this can be merged.
>
> Thanks for the work though.
>
> ------------------------------------------------------------------------
>
> In source/nanoFramework.Json.sln 
> <https://github.com/nanoframework/lib-nanoFramework.Json/pull/23#discussion_r450107817>:
>
> >  Project("{11A8DD76-328B-46DF-9F39-F559912D0360}") = "nanoFramework.Json.Test", "nanoFramework.Json.Test\nanoFramework.Json.Test.nfproj", "{65B0ED92-3D3C-4289-A267-4A4B160FCF13}"
>
>   EndProject
>
> -Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution Items", "{BAFE1F60-FD27-41B0-9E8C-117F56001A8C}"
>
> -	ProjectSection(SolutionItems) = preProject
>
> -		nanoFramework.Json.nuspec = nanoFramework.Json.nuspec
>
> -		NuGet.Config = NuGet.Config
>
> -		version.json = version.json
>
> -	EndProjectSection
>
> +Project("{11A8DD76-328B-46DF-9F39-F559912D0360}") = "nanoFramework.Json", "nanoFramework.Json\nanoFramework.Json.nfproj", "{181141FB-0D05-41C8-8692-81A2989E6B4B}"
>
> Nuget Spec is required, not sure why it has been deleted...
>
> ------------------------------------------------------------------------
>
> In source/nanoFramework.Json.Test/app.config 
> <https://github.com/nanoframework/lib-nanoFramework.Json/pull/23#discussion_r450108587>:
>
> > @@ -0,0 +1,11 @@
>
> +﻿<?xml version="1.0" encoding="utf-8"?>
>
> +<configuration>
>
> +  <runtime>
>
> +    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
>
> +      <dependentAssembly>
>
> +        <assemblyIdentity name="mscorlib" publicKeyToken="c07d481e9758c731" culture="neutral" />
>
> +        <bindingRedirect oldVersion="0.0.0.0-1.7.3.5" newVersion="1.7.3.5" />
>
> app.config files are bad, it shows that there is an issue with the 
> nugets installed. Try removing the file and ensure it still builds 
> after updating the nugets.
>
> ------------------------------------------------------------------------
>
> In source/nanoFramework.Json.Test/nanoFramework.Json.Test.nfproj 
> <https://github.com/nanoframework/lib-nanoFramework.Json/pull/23#discussion_r450108739>:
>
> >      </Reference>
>
>     </ItemGroup>
>
>     <ItemGroup>
>
> +    <None Include="app.config" />
>
> app.config files are bad, it shows that there is an issue with the 
> nugets installed. Try removing the file and ensure it still builds 
> after updating the nugets.
>
> ------------------------------------------------------------------------
>
> In source/nanoFramework.Json/app.config 
> <https://github.com/nanoframework/lib-nanoFramework.Json/pull/23#discussion_r450109065>:
>
> > @@ -0,0 +1,11 @@
>
> +﻿<?xml version="1.0" encoding="utf-8"?>
>
> +<configuration>
>
> +  <runtime>
>
> +    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
>
> +      <dependentAssembly>
>
> +        <assemblyIdentity name="mscorlib" publicKeyToken="c07d481e9758c731" culture="neutral" />
>
> +        <bindingRedirect oldVersion="0.0.0.0-1.7.3.5" newVersion="1.7.3.5" />
>
> app.config files are bad, it shows that there is an issue with the 
> nugets installed. Try removing the file and ensure it still builds 
> after updating the nugets.
>
> ------------------------------------------------------------------------
>
> In source/nanoFramework.Json/nanoFramework.Json.nfproj 
> <https://github.com/nanoframework/lib-nanoFramework.Json/pull/23#discussion_r450109194>:
>
> >      </Reference>
>
>     </ItemGroup>
>
>     <ItemGroup>
>
> +    <None Include="app.config" />
>
> app.config files are bad, it shows that there is an issue with the 
> nugets installed. Try removing the file and ensure it still builds 
> after updating the nugets.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nanoframework/lib-nanoFramework.Json/pull/23#pullrequestreview-442929561>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AO2PVOK7N2NR3Y37ZW6HUB3R2GND5ANCNFSM4OQPP2SQ>.
>
